engine/order_manager.go

```diff
@@ -589,6 +589,14 @@ func (m *OrderManager) processSubmittedOrder(newOrderResp *order.SubmitResponse)
 		return nil, err
 	}
 
+	if err := m.orderStore.add(detail.CopyToPointer()); err == ErrOrdersAlreadyExists {
```

Comment:
Please use `errors.Is(...)`. We wrap errors and using `==` will not be able to handle that. Even if its fine now, better to properly handle it now than to rely on this and have it fail silently when it doesn't need to

```go
func TestError(t *testing.T) {
	err1 := fmt.Errorf("%w hello", errEmptyOrderIDs)
	if err1 != errEmptyOrderIDs {
		t.Fatal("fail")
	}
}
```

```go
func TestWrapError(t *testing.T) {
	err1 := fmt.Errorf("%w hello", errEmptyOrderIDs)
	if !errors.Is(err1, errEmptyOrderIDs) {
		t.Fatal("fail")
	}
}
```

Reply:
Yep, completely agree